File: solutions/lua/robot-name/1/robot-name.lua
local Robot = {}

local function generate_name()
    local name = ""
    name = name .. string.char(math.random(65, 90))
    name = name .. string.char(math.random(65, 90))
    name = name .. math.random(0, 9)
    name = name .. math.random(0, 9)
    name = name .. math.random(0, 9)
    return name
end

function Robot:new()
    local robot = {}

    setmetatable(robot, self)
    self.__index = self

    robot.name = generate_name()

    return robot
end

function Robot:reset()
    self.name = generate_name()
end

return Robot

File: solutions/python/robot-name/1/robot_name.py
import random


class Robot:

    def __random_letter():
        return chr(random.randint(ord('A'), ord('Z')))

    def __random_digit():
        return str(random.randint(0, 9))

    def __create_name():
        random.seed()
        name = [ Robot.__random_letter(), Robot.__random_letter(),
                 Robot.__random_digit(), Robot.__random_digit(), Robot.__random_digit() ]
        return "".join(name) 
    
    def __init__(self):
        self.name = Robot.__create_name()

    def reset(self):
        self.name = Robot.__create_name()
